server/src/main/java/org/opensearch/action/resync/TransportResyncReplicationAction.java

```diff
@@ -109,7 +111,8 @@ public TransportResyncReplicationAction(
             EXECUTOR_NAME_FUNCTION,
             true, /* we should never reject resync because of thread pool capacity on primary */
             indexingPressureService,
-            systemIndices
+            systemIndices,
+            tracer
```

Comment:
Will passing tracer publish the spans for these actions as well? If yes, is this expected?

Reply:
Just passing the tracer instance wont do anything. If we are creating the span in the generic code of a parent class then it anyways would generate the spans for all the implementations since parent is having the tracer instance.

Reply:
Yes, the methods in parent class are adding spans. Given these concrete actions may not be useful from tracing perspective for those methods when executed in the context of these actions, I wanted to evaluate if passing a NoOpTracer to parent class through these concrete actions may be a better idea.

Reply:
@Gaganjuneja Spans were being added as we were creating them in the parent class - TransportWriteAction ( we were passing the injected tracer to TransportWriteAction). 
Agree @mgodwan , while the spans created won't have much impact, it is correct that they do not add value currently. 
Hence passed a NoopTracer instance from the base classes where spans are not required to avoid creating them

Reply:
Yes it may reduce data but then it will always be a code change to enable/disable traces for certain code paths etc. It should have controlled through sampling. You mat think it in terms of logging we can enable/disable through appender settings. Just give it a thought.

Reply:
I would say this is expected in general - with instrumenting the base write action we will cover a bit more than just indexing path. This has to be done sooner or later, I don't think we should circumvent that.

Reply:
> It should have controlled through sampling. You mat think it in terms of logging we can enable/disable through appender settings. Just give it a thought.

@Gaganjuneja It skips publishing those trace but still may add memory/compute overhead for those, right, or will it ignore traces completely (based on some attribute)? Is there any code pointer/documentation on how traces can be controlled through appender settings? It will be useful to understand this further.

> This has to be done sooner or later, I don't think we should circumvent that.

If we have sampling strategy/settings to allow control on this, I don't see any concerns with injecting tracer for these actions. Otherwise, I believe it may be good to enable tracing for actions/specific code flows as needed.

Reply:
@mgodwan @rayshrey the sampling strategy is there already: we sample 1% of requests at the moment by default when the tracing feature is enabled.

Reply:
Got it. Thanks @reta @rayshrey for the confirmation.

Reply:
I was talking about component level Sampling.

Reply:
> I was talking about component level Sampling.

Thanks @Gaganjuneja  we have https://github.com/opensearch-project/OpenSearch/issues/8918 - let collect the feedback there.